statics/js/components/graph-layout.js

```diff
@@ -1,6 +1,187 @@
+        return 0;
+      }
+
+      return Math.floor(8 * totalByte * 1000 / deltaMillis); // bits-per-second 
+    },
+
+    setup: function(topology, link) {
+      const defaultBandwidthBaseline = 1024 * 1024 * 1024; // 1 gbps
+
+      this.link = link;
+
+      var bandwidthBaseline = (topology.bandwidth.bandwidthThreshold === 'relative') ?
+        link.target.metadata.Speed || defaultBandwidthBaseline : 1;
+
+      this.bandwidth = topology.bandwidth;
+      this.bandwidthAbsolute = this.bandwidthFromMetrics(link.target.metadata.LastUpdateMetric);
+      this.bandwidthCheck = this.bandwidthAbsolute / bandwidthBaseline;
+    },
+
+    hasData: function() {
+      if (!this.link.target.metadata.LastUpdateMetric) {
+        return false;
+      }
+
+      return this.bandwidthCheck > this.bandwidth.active;
+    },
+
+    getText: function() {
+      return bandwidthToString(this.bandwidthAbsolute);
+    },
+
+    isActive: function() {
+      return (this.bandwidthCheck > this.bandwidth.active) && (this.bandwidthCheck < this.bandwidth.warning);
+    },
+
+    isWarning: function() {
+      return (this.bandwidthCheck >= this.bandwidth.warning) && (this.bandwidthCheck < this.bandwidth.alert);
+    },
+
+    isAlert: function() {
+      return this.bandwidthCheck >= this.bandwidth.alert;
+    },
+  },
+});
+
+var LinkLabelLatency = Vue.extend({
+  mixins: [apiMixin],
+
+  methods: {
+    setup: function(topology, link) {
+      this.active = 0;
+      this.warning = 10
+      this.alert = 100;
+      this.link = link;
+      this.updateData();                    // prepare data for next cycle
+    },
+
+    flowKey: function(a) {
+      return `TrackingID:${a.TrackingID},ANodeTID:${a.ANodeTID},BNodeTID:${a.BNodeTID}`;
+    },
+
+    linklabel: function(a, b) {
+      return `${a.Name}-->${b.Name}`;
+    },
+
+    flowCompare: function(a, b) {
+      keya = this.flowKey(a);
+      keyb = this.flowKey(b);
+      return keya.localeCompare(keyb);
+    },
+
+    updateLatency: function(flows) {
+      let prev, curr;
+      for (i in flows) {
+        prev = curr;
+        curr = flows[i];
+
+        if (!prev) {
+          continue;
+        }
+
+        if (self.flowKey(curr) != self.flowKey(prev)) {
+          continue;
+        }
+
+        latency = Math.abs(curr.Start - prev.Start);
+        this.link.latencyTimestamp = Math.max(curr.Start, prev.Start);
+        this.link.latency = latency;
+        return true;
+      }
+      return false;
+    },
+
+    updateData: function() {
+      self = this;
+
+      const a = this.link.source.metadata;
+      const b = this.link.target.metadata;
+      const prefix = `${a.Name}<-->${b.Name}`
+
+      if (!a.Capture) {
+        return;
+      }
+      if (!b.Capture) {
+        return;
+      }
+
+      const limit = 10;
+      const anyNode = 'Within("'+a.TID+'", "'+b.TID+'")';
+      const query = 'G.Flows().Has("NodeTID", '+anyNode+').Limit('+limit+')';
```

Comment:
not sure this will be accurate to compare 'start' and not RTT

Reply:
ok have changed to `RTT` and set timestamp using `Last`